### DISCUSSION TITLE
Support `peer` and `group` with custom variants

### DISCUSSION THREAD

**Discussion by TastyPi** [tailwindlabs/tailwindcss#7364] [Ideas]
It seems the `peer` and `group` modifiers don't work with custom variants (unless I'm missing something?).

For example, it would be nice to be able to do this:

```javascript
addVariant('expanded', '&[aria-expanded="true"]')
```

```html
<button aria-expanded="false">Expand</button>
<div class="hidden peer-expanded:block">Hello World</div
```

but that does not currently seem possible without defining `peer-expanded` manually.

**Comment by adamwathan** [comment]
Yeah have to define manually, we define all the built-in ones manually too. Agree would be cool to figure out a way to make it a bit more automatic if possible.

**Comment by sntran** [comment]
From the sources, looks like you can add another variant for peer and group:

```js
addVariant("group-expanded", `:merge(.group)[aria-expanded="true"] &`)
addVariant("peer-expanded", `:merge(.peer)[aria-expanded="true"] ~ &`);
```

It's not automagically, but it's only 2 more lines to add to your plugin.

**Reply by TastyPi** [reply]
@sntran yup, and I have done that, but only after an hour of trying to figure out why it wasn't working 🙂

**Comment by sergio-agosti** [comment]
@adamwathan @TastyPi I always add a small utility function to do that in one go:

```javascript
const addVariantWithStates = (variant, selector) => {
    addVariant(variant, `&${selector}`)
    addVariant(`group-${variant}`, `:merge(.group)${selector} &`)
    addVariant(`peer-${variant}`, `:merge(.peer)${selector} ~ &`)
}
```

**Reply by sergio-agosti** [reply]
Usage example:

```javascript
addVariantWithStates('expanded', '[aria-expanded="true"]')
```
